setup.py

```diff
@@ -1,17 +0,0 @@
-# -----------------------------------------------------------------------------
-#
```

Comment:
Can we keep this to have backward compatibility going? `pip install -e .` or `python setup.py install`

Reply:
With this minimal setup.py, it installs with package name "UNKNOWN". Otherwise, we have to repeat all the info in `pyproject.toml` in this file also. As per: [setup.py deprecation](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/), it's not recommended to run `python setup.py` commands anymore. Not having setup.py also ensures this.

Reply:
removing setup.py, doesn't ensure `pip install -e .` works right?

Reply:
setup.py has nothing to do with pip. Even after removing setup.py, I'm able to do editable install with `pip install -e .`.

Reply:
Okay, i was facing the issue to do it, let me retest it once.

Reply:
I've added minimum version for setuptools to support pyproject.toml only project. `pip install .` or `pip install git+<repo_url>` should work without issues when pip is reasonably up-to-date.